oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/MergeRunner.java

```diff
@@ -297,14 +332,20 @@ private class Task implements Callable<File> {
             mergeTaskPhaser.register();
         }
 
+
         @Override
-        public File call() {
+        public File call() throws Exception {
             try {
-                if (merge(mergeTarget, mergedFile)) {
-                    log.info("merge complete for {}", mergedFile.getName());
-                    return mergedFile;
+                String mergedFileName = mergedFile.getName();
+                if (mergeCancelled.get()) {
+                    log.debug("merge cancelled, skipping merge task");
+                    throw new Exception("merge skipped for " + mergedFileName);
```

Comment:
It would be good to use a different exception here, for example java.io.EOFException

Reply:
changed